pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -49,13 +51,17 @@ type Controller struct {
 	podListerSynced cache.InformerSynced
 
 	queue workqueue.RateLimitingInterface
+
+	// allows overriding for testing
+	features utilfeature.FeatureGate
```

Comment:
This isn't appropriate here. Rather than tying yourself to CLI related feature on/off here, you can add a specific bool for your specific need and use it.  Then establish your plumbing.